src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1127,23 +1122,20 @@ class SQLTargetWHandle : public StorageTargetWHandle {
     }
 
     row_id_ = sqlite3_last_insert_rowid(db_.get());
-    db_.commitTransaction();
   }
 
   ~SQLTargetWHandle() override {
     if (!closed_) {
       LOG_WARNING << "Handle for file " << target_.filename() << " has not been committed or aborted, forcing abort";
+      db_.commitTransaction();
```

Comment:
Didn't notice it before, but why do we call commit here? We want to abort the transaction, right?

Reply:
Not quite sure actually. My idea was that if we don't abort explicitly then we probably want to save what we've already downloaded.
By the way, maybe we don't need explicit `pause` from `Aktualizr::Shutdown` in the end, @patrickvacek's commit confused me somewhat. Just need to add the test that the file really gets committed.

Reply:
@OYTIS good call, actually! I overlooked the commit in the destructor, which is actually an even better solution than the implicit call to pause.